lib/core/filtering/simple_rule.rb

```diff
@@ -0,0 +1,144 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'utility/logger'
+
+module Core
+  module Filtering
+    class SimpleRule
+      POLICY = 'policy'
+      FIELD = 'field'
+      RULE = 'rule'
+      VALUE = 'value'
+      ID = 'id'
+
+      DEFAULT_RULE_ID = 'DEFAULT'
+
+      class Policy
+        INCLUDE = 'include'
+        EXCLUDE = 'exclude'
+      end
+
+      class Rule
+        REGEX = 'regex'
+        EQUALS = 'equals'
+        STARTS_WITH = 'starts_with'
+        ENDS_WITH = 'ends_with'
+        CONTAINS = 'contains'
+        LESS_THAN = '<'
+        GREATER_THAN = '>'
+      end
+
+      attr_reader :policy, :field, :rule, :value, :id
+
+      def initialize(rule_hash)
+        @policy = rule_hash.fetch(POLICY)
+        @field = rule_hash.fetch(FIELD)
+        @rule = rule_hash.fetch(RULE)
+        @value = rule_hash.fetch(VALUE)
+        @id = rule_hash.fetch(ID)
+        @rule_hash = rule_hash
+      rescue KeyError => e
+        raise "#{e.key} is required"
+      end
+      def self.from_args(id, policy, field, rule, value)
+        SimpleRule.new(
+          {
+            ID => id,
+            POLICY => policy,
+            FIELD => field,
+            RULE => rule,
+            VALUE => value
+          }
+        )
+      end
+
+      DEFAULT_RULE = SimpleRule.new(
+        'policy' => 'include',
+        'field' => '_',
+        'rule' => 'regex',
+        'value' => '.*',
+        'id' => SimpleRule::DEFAULT_RULE_ID
+      )
+
+      def match?(document)
+        return true if id == DEFAULT_RULE_ID
+        doc_value = document[field]
+        return false if doc_value.nil?
+        coerced_value = coerce(doc_value)
```

Comment:
Why in the code below the `doc_value` is sometimes compared with `value` and sometimes with `coerced_value`?

Also what is a `coerced_value`?